spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2501,6 +2501,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
       case SortExec(sortOrder, _, child, _)
           if isCometOperatorEnabled(op.conf, CometConf.OPERATOR_SORT) =>
+        // TODO: Remove this constraint when we upgrade to new arrow-rs including
+        // https://github.com/apache/arrow-rs/pull/6225
+        if (child.output.length == 1 && child.output.head.dataType.isInstanceOf[StructType]) {
```

Comment:
As we add support for other types, do we need to update this to make it recursive so that we check for Map or Array containing struct?

Reply:
Let me add more data types here according to arrow-rs.